lib/puppet/util/platform.rb

```diff
@@ -33,6 +33,22 @@ def fips_enabled?
       def self.jruby?
         RUBY_PLATFORM == 'java'
       end
+
+      def jruby_fips?
+        @@jruby_fips ||= if RUBY_PLATFORM == 'java'
+                           require 'java'
+
+                           begin
+                             require 'openssl'
+                             false
```

Comment:
In MRI ruby you can ask openssl if fips is enabled:

```
$ ruby -ropenssl -e "puts OpenSSL::OPENSSL_FIPS"
false
```

Should we do the same for JRuby to future-proof jruby+fips being able to load openssl?

Reply:
You can ask JRuby's OpenSSL implementation if it's fips enabled (it's hardcoded to false), but the way we've done our FIPS builds you can't load it's OpenSSL to ask it that question.

Reply:
😆 😭